assets/js/app.js

```diff
@@ -1,122 +1,147 @@
+	 * @param  {String} character [description]
+	 * @param  {String} cardArea  [description]
+	 * @return {[type]}           [description]
+	 */
+	function createCharacterCard(character, cardArea){
+		// Create a div for each character's card containing name, healthPoints, and image.
+		const $characterDiv = $("<div>", {"class": "characters"})
+		.attr("data-name", character.name);
+		const $characterName = $("<p>")
+		.attr("class", "characterName")
+		.text(character.name);
+		const $characterImage = $("<img>")
+		.attr("class", "characterImage")
+		.attr("src", character.image);
+		const $characterHealthPoints = $("<p>")
+		.attr("class", "characterHealthPoints")
+		.text(character.healthPoints);
+
+		$characterDiv
+		.append($characterName)
+		.append($characterHealthPoints)
+		.append($characterImage);
+
+		$(cardArea).append($characterDiv);
+	};
+
+	/**
+	 * [setCharacterSelection description]
+	 * @param {[type]} enemyArray [description]
+	 */
+	function setCharacterSelection (enemyArray) {
+		console.log("setCharacterSelection function!");
+		enemyArray.forEach(function(enemyCard) {
+			console.log("move the enemies!");
+			console.log(enemyCard);
+			createCharacterCard(enemyCard, "#enemyArea");
+			console.log("enemies moved!");
+			$("#header-text").text("Your Character");
+		})
+	};
+	/**
+	 * [init description]
+	 * @return {[type]} [description]
+	 */
+	function init() {
+		createCharacter();
+	};
+
+	return {
+		createCharacter,
+		createCharacterCard,
+		defender: '',
+		enemies : [],
+		init,
+		setCharacterSelection,
+		yourCharacter: ''
 	}
-}
-
-createCharacter();
-
-// Now I need to declare my variables and create an onclick event for the character cards so that the selected character remains and the rest gets pushed into an array and appends to the 'enemies to attack' div. 
-let yourCharacter;
-const enemies = [];
-let defender; // this is the opponent
-// const healthPoints;
-// const enemyAttackBackPoints;
-
-
-function setCharacterSelection (enemyArray) {
-	console.log("setCharacterSelection function!");
-	enemyArray.forEach(function(enemyCard) {
-		console.log("move the enemies!");
-		console.log(enemyCard);
-		createCharacterCard(enemyCard, "#enemyArea");
-		console.log("enemies moved!");
-		$("#header-text").text("Your Character");		
-	})
-};
 
+}())
+
+
+// Executes code when DOM has fully loaded
+$(document).ready(function() {
+	character.init()
+
+// Create the character object to hold the characters data
 
 $("#characterArea").on("click", ".characters", function() {
 	console.log("character selected...");
 	// Stores clicked character's name
 	const name = $(this).attr("data-name");
 	console.log(name);
 	// Runs if your player has not been selected
-	if(!yourCharacter) {
+	if(!character.yourCharacter) {
 
-		yourCharacter = name;
-		console.log(`Your character is ${yourCharacter}`);
+		character.yourCharacter = name;
+		console.log(`Your character is ${character.yourCharacter}`);
 
 		for(var key in characters) {
```

Comment:
`for...in` vs `for...of` vs `$.each`

Also, this click event will continually append the enemies of the person you clicked on in the first iteration. Meaning that you keep adding the same enemies over and over again. You should have a reset function somewhere that delete all the div's in the #enemies are and re-appends the updated array.

Reply:
could I also add `break` so that it closes the iterator and triggers return?